frank-doc-frontend/src/app/pipes/as-text.pipe.ts

```diff
@@ -9,33 +9,39 @@ export class AsTextPipe implements PipeTransform {
     if (!value) return "";
     value = value.replaceAll('\\"', '"');
     value = value.replace(/<[^>]*>?/gm, '');
-    value = value.replace(/(?:{@link\s(.*?)})/g, function (match, captureGroup) {
-      // {@link PipeLineSession pipeLineSession}
-      // {@link IPipe#configure()}
-      // {@link #doPipe(Message, PipeLineSession) doPipe}
-      let referencedElement = captureGroup;
-      const hash = captureGroup.indexOf("#");
-      if (hash > -1) {
-        referencedElement = captureGroup.split("#")[0];
+    value = value.replace(/(?:{@link\s(.*?)})/g, (match, captureGroup: string) => {
+      // {@link PipeLineSession pipeLineSession} -> 'PipeLineSession pipeLineSession'
+      // {@link IPipe#configure()} -> 'IPipe#configure()'
+      // {@link #doPipe(Message, PipeLineSession) doPipe} -> '#doPipe(Message, PipeLineSession) doPipe'
+      const hashPos = captureGroup.indexOf("#"),
+        isMethod = hashPos > -1,
+        element = isMethod ? captureGroup.split("#")[0] : captureGroup;
 
-        if (referencedElement == '') { //if there is no element ref then it's a method
-          const method = captureGroup.slice(hash);
-          const nameOrAlias = method.split(") ");
-          if (nameOrAlias.length == 2) {
-            return nameOrAlias[1]; //If it's an alias
-          }
-          return method.slice(1, method.indexOf("("));
-        }
+      if (element == '') { //if there is no element ref then it's a method
+        return this.transformInternalMethod(captureGroup, hashPos);
       }
-      const captures = referencedElement.split(" "); //first part is the ClassName, 2nd part the written name
-      let name = captures[captures.length - 1];
-      if (hash > -1) {
-        const method = captureGroup.split("#")[1];
-        name = name + "." + (method.slice(method.indexOf(") ") + 1)).trim();
-      }
-      return name;
+
+      const elementParts = element.split(" "); //first part is the class name, 2nd part the written name
```

Comment:
Als je al weet wat er in zit kun je het prima al aan een naam geven, dan hoef je later niet meer `elementParts[0]` te doen, maar `className`
```TypeScript
const [className, writtenName] = element.split(" ");
```

Reply:
In deze context is het niet zeker of er een tweede index uit komt, helaas kan dit dus niet

Reply:
Natuurlijk kan dat wel. Dan is de tweede undefined. 
Dan hoef je dus ook niet te checken op lengte van een array of dat soort dingen, dan is het een makkelijkere null check.